### ISSUE TITLE
bad discovery cluster: <nil>

### ISSUE THREAD

**Issue by colebrumley** [coreos/etcd#2319]
When testing the latest release (2.0.3) for the issue reported in #2297, I am seeing a new error:

```
Feb 16 08:54:26 VA1ACHMVP01 systemd[1]: Unit etcd.service entered failed state.
Feb 16 08:54:26 VA1ACHMVP01 systemd[1]: etcd.service: main process exited, code=exited, status=1/FAILURE
Feb 16 08:54:26 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:26 etcd: bad discovery cluster: <nil>
Feb 16 08:54:26 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:26 etcd: stopping listening for peers on http://10.126.75.72:7001
Feb 16 08:54:26 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:26 etcd: stopping listening for peers on http://127.0.0.1:7001
Feb 16 08:54:26 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:26 etcd: stopping listening for client requests on http://10.126.75.72:4001
Feb 16 08:54:26 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:26 etcd: stopping listening for client requests on http://127.0.0.1:4001
Feb 16 08:54:26 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:26 discovery: found 2 needed peer(s)
Feb 16 08:54:26 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:26 discovery: found peer c941b1bd3872caf in the cluster
Feb 16 08:54:10 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:10 discovery: found 1 peer(s), waiting for 1 more
Feb 16 08:54:10 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:10 discovery: found self 9674d30268039d3e in the cluster
Feb 16 08:54:07 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:54:07 discovery: during waiting for other nodes connection to https://discovery.etcd.io timed out, retrying in 2s
Feb 16 08:53:08 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:53:08 discovery: found 1 peer(s), waiting for 1 more
Feb 16 08:53:08 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:53:08 discovery: found self 9674d30268039d3e in the cluster
Feb 16 08:53:07 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:53:07 etcd: listening for client requests on http://127.0.0.1:4001
Feb 16 08:53:07 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:53:07 etcd: listening for client requests on http://10.126.75.72:4001
Feb 16 08:53:07 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:53:07 etcd: listening for peers on http://127.0.0.1:7001
Feb 16 08:53:07 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:53:07 etcd: listening for peers on http://10.126.75.72:7001
Feb 16 08:53:07 VA1ACHMVP01 etcd[10602]: 2015/02/16 08:53:07 no data-dir provided, using default data-dir ./mvp01.etcd
Feb 16 08:53:07 VA1ACHMVP01 systemd[1]: Started Etcd Server.
Feb 16 08:53:07 VA1ACHMVP01 systemd[1]: Starting Etcd Server...
```

This is a new cluster using the public discovery service.  No data-dir is created before the application dies, so I have no logs to provide.  This is a 2-node test cluster in case that makes a difference.

**Comment by xiang90** [comment]
@colebrumley Can you provide the command line arguments you gave to the etcd members (all of them)?

**Comment by colebrumley** [comment]
I called etcd from a systemd unit with the following args:

```
/usr/bin/etcd \
 -advertise-client-urls=http://${CLUSTER_IP}:4001,http://127.0.0.1:4001 \
 -listen-client-urls=http://${CLUSTER_IP}:4001,http://127.0.0.1:4001 \
 -initial-advertise-peer-urls=http://${CLUSTER_IP}:7001,http://127.0.0.1:7001 \
 -listen-peer-urls=http://${CLUSTER_IP}:7001,http://127.0.0.1:7001 \
 -name=${NODE_NAME} \
 -discovery=${DISCOVERY}
```

The only differences between the two nodes are the NODE_NAME and CLUSTER_IP

**Comment by xiang90** [comment]
@colebrumley You gave duplicated `advertise-peer-url` across the cluster. 
Please check the doc at https://github.com/coreos/etcd/blob/master/Documentation/configuration.md#-initial-advertise-peer-urls.

We will improve the error message.

**Comment by colebrumley** [comment]
Ah I see it now, thanks for the quick reply!

**Comment by xiang90** [comment]
@colebrumley The wrong error message is also fixed at #2320.
